jbpm-wb-common/jbpm-wb-common-client/src/main/java/org/jbpm/workbench/common/client/util/UTCTimeBoxImplShared.java

```diff
@@ -60,26 +60,28 @@ public void validate() {}
     @Override
     public final void setValue(Long value) {
         setValue(value, false);
-    }    
-    
+    }
+
     // ----------------------------------------------------------------------
-    
+
     public UTCTimeBoxImplShared() {
         if (fallbackTimeFormats == null) {
-            fallbackTimeFormats = new DateTimeFormat[fallbackFormatStrings.length];
-            for (int i=0; i<fallbackFormatStrings.length; i++) {
-                fallbackTimeFormats[i] = DateTimeFormat.getFormat(fallbackFormatStrings[i]);
+            // Initialize the array fully in local var before assigning to class field to avoid synchronization issues
```

Comment:
findbugs plugin warning to clarify this fix:

UTCTimeBoxImplShared.java:68, LI_LAZY_INIT_UPDATE_STATIC, Priority: High

Incorrect lazy initialization and update of static field org.jbpm.workbench.common.client.util.UTCTimeBoxImplShared.fallbackTimeFormats in new org.jbpm.workbench.common.client.util.UTCTimeBoxImplShared()

This method contains an unsynchronized lazy initialization of a static field. After the field is set, the object stored into that location is further updated or accessed. The setting of the field is visible to other threads as soon as it is set. If the futher accesses in the method that set the field serve to initialize the object, then you have a very serious multithreading bug, unless something else prevents any other thread from accessing the stored object until it is fully initialized.

Even if you feel confident that the method is never called by multiple threads, it might be better to not set the static field until the value you are setting it to is fully populated/initialized.